File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/LessOrEqualThanValidator.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.base.comum.HelperDate;
import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.base.comum.validadores.CustomValidator;
import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.LessOrEqualThan;
import java.util.Date;

public class LessOrEqualThanValidator
extends CustomValidator<LessOrEqualThan> {
    private static final String COMPARED_KEY = "1";

    @Override
    public Object getParameter(String key) {
        if (key.equals(COMPARED_KEY)) {
            return ((LessOrEqualThan)this.getAnnotation()).value();
        }
        return null;
    }

    @Override
    public String getMessage() {
        return ((LessOrEqualThan)this.getAnnotation()).message();
    }

    @Override
    public byte[] getGroups() {
        return ((LessOrEqualThan)this.getAnnotation()).groups();
    }

    @Override
    protected String getCondition() {
        return ((LessOrEqualThan)this.getAnnotation()).condition();
    }

    @Override
    public boolean isValid(ValidatorContext context) {
        Object value1 = context.getValue();
        Object value2 = context.getMemberValue(((LessOrEqualThan)this.getAnnotation()).value());
        if (Utils.nulo(value1) || Utils.nulo(value2)) {
            return true;
        }
        return HelperDate.dateAfterOrEquals((Date)value2, (Date)value1);
    }
}

File: src/lib/pjecalc/core/comum/validators/less-or-equal-than-validator.ts
/**
 * PJe-Calc v2.15.1 — LessOrEqualThanValidator
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.LessOrEqualThanValidator
 */
import { HelperDate } from '../../base/comum/helper-date';
import { CustomValidator } from './custom-validator';
import type { LessOrEqualThan_ } from './less-or-equal-than';
import type { ValidatorContext } from './validator-context';

export class LessOrEqualThanValidator extends CustomValidator<LessOrEqualThan_> {
  private static readonly COMPARED_KEY = '1';

  getMessage(): string { return this.spec.message; }
  getGroups(): number[] { return this.spec.groups; }
  protected getCondition(): string { return this.spec.condition; }

  getParameter(key: string): unknown {
    if (key === LessOrEqualThanValidator.COMPARED_KEY) return this.spec.value;
    return null;
  }

  isValid(context: ValidatorContext): boolean {
    const value1 = context.getValue();
    const value2 = context.getMemberValue(this.spec.value);
    if (value1 == null || value2 == null) return true;
    if (value1 instanceof Date && value2 instanceof Date) {
      return HelperDate.dateAfterOrEquals(value2, value1);
    }
    return true;
  }
}
